src/xunit.core/MemberDataAttributeBase.cs

```diff
@@ -66,11 +67,11 @@ public override IEnumerable<object[]> GetData(MethodInfo testMethod)
             if (obj == null)
                 return null;
 
-            var dataItems = obj as IEnumerable<object>;
+            var dataItems = obj as IEnumerable;
             if (dataItems == null)
                 throw new ArgumentException($"Property {MemberName} on {type.FullName} did not return IEnumerable<object>");
```

Comment:
Nit: Exception message stills says `IEnumerable<object>`, should now just be `IEnumerable`